### ISSUE TITLE
Only one project on workspace but evaluation still ask user to specify projectName in launch.json

### ISSUE THREAD

**Issue by testforstephen** [Microsoft/vscode-java-debug#205]
Debugging project https://github.com/vscjavaci/java-debug-ci/tree/master/testcase/4.variable, watching variables always ask user to `specify projectName in launch.json`. But the workspace only contains one project, the debugger should automatically take this project as evaluation project.

**Comment by leandro-ss** [comment]
I have de same problem,.

Do you have a example? TOdo??

**Comment by andxu** [comment]
https://github.com/Microsoft/java-debug/pull/146

**Comment by andxu** [comment]
@leandro-ss , find the project name in .project or delete the launch.json and regenerate it.

**Comment by aboe026** [comment]
Where do we find the projectName? I tried just using the name of the folder (the folder I selected in `File -> Open Folder...` when starting vscode) but it said `Project <root-workspace-folder cannot be found`.

**Comment by yaohaizh** [comment]
@aboe026 Please see the comment above

**Comment by aboe026** [comment]
I don't see any `.project` file, and when i delete the launch.json and regenerate it, it does not contain the projectName property:
```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "type": "java",
            "name": "Debug (Attach)",
            "request": "attach",
            "hostName": "localhost",
            "port": 0
        }
    ]
}
```
I generate the launch.json by going to the Debug Icon on the left, clicking the dropdown that says "No Configurations", then selecting "Add Configuration..." and choosing "Java"

**Comment by zhanglid** [comment]
I have the same issue

**Comment by andxu** [comment]
@aboe026 , you have not the launch-able class[ class with the static void main(String[]args) method]  in your project,, if you want to debug the program in launch mode, you need to create a main class, and then press f5, if you want to debug it with the attach mode, you need to setup a java project first using maven/gradle or eclipse.

**Comment by andxu** [comment]
Fixed in 0.7.0, for attaching mode, it will be fixed at 0.8.0(another issue will be opened tracking attach mode evaluation)

**Comment by aboe026** [comment]
@andxu what is the issue number for fixing it in attach mode? I see the latest release is `0.9.0` but I'm still getting the same error. Has it still not been fixed for attach mode?

**Comment by andxu** [comment]
@aboe026 it is supposed to be fixed at 0.7.0 (2018-3-15), but in some corner cases(multiple-root), the projectName is still needed, my fix only takes a lot of tries to guess out what the correct projectName, for multiple projects, it will guess based on the current class stopped, if multiple projects contains this class, then the evaluation will fail with the error message `specify projectName in launch.json `.

**Comment by balazsmiklos85** [comment]
With 1.45.1 I still get the same error message, none of the suggestions above help, and I don't have multiple projects. (Or at least I'm not aware of it. What exactly qualifies as a "project"?)
